src/WebJobs.Script.WebHost/Management/LinuxInstanceManager.cs

```diff
@@ -139,8 +103,24 @@ private bool IsValidEnvironment(HostAssignmentContext context)
 
         private async Task AssignAsync(HostAssignmentContext assignmentContext)
         {
+            if (assignmentContext.IsWarmupRequest)
```

Comment:
As mentioned in the other comment, what you had before is preferred (keeping this before the lock statements in `StartAssignment` and `AssignInstanceAsync`.)